docs/modules/schema-validation/con-record-validation.adoc

```diff
@@ -0,0 +1,83 @@
+// file included in the following:
+//
+// index.adoc
+
+[id='con-record-validation-{context}']
+= Record Validation
+
+[role="_abstract"]
+The Record Validation filter validates records sent by a producer.   Only records that pass the validation are sent to
+the broker. This filter can be used to prevent poison messages entering the Kafka system, that may otherwise lead to
+consumer failure.
+
+The filter currently supports two modes of operation:
+
+1. Schema validation validates the content of the record against a schema stored in an https://www.apicur.io/registry/[Apicurio Registry].
+2. Syntactically Correct JSON validation ensures the producer is producing messages that contain syntactically valid JSON.
+
+Validation rules can validate the content of the Kafka record key or value.
+
+If the validation fails, the product request is rejected and the producing application receives an error response.  The broker
+will not receive the rejected records.
+
+== How to use the filter
+
+There is just one step to use the filter.
+
+1. <<Configuring virtual clusters>>
+2. Configuring the filter within Kroxylicious.
+
+=== Configuring the filter within Kroxylicious.
+
+.Example Record Validation filter configuration
+
+[source,yaml]
+----
+filters:
+  - type: ProduceValidationFilterFactory
+    config:
+        rules:
+        - topicNames:                                            # <1>
+            - <topic name>
+            keyRule:
+              <rule definition>                                  # <2>
+            valueRule:
+              <rule definition>                                  # <3>
+        isForwardPartialRequests: false                          # <4>
```

Comment:
unrelated to this PR but thats a horrible name for a config entry.

Reply:
We could use an Jackson alias to fix that.

Reply:
The field name is actual correct. It is just the documentation that is wrong.